### PR TITLE
Add ResourceName interface methods to all implementing classes

### PR DESCRIPTION
**Pull request by andreamlin**

### CODE REVIEW

File: plugin/templates/resource_name_fixed.mustache [hunk 1]
```diff
@@ -16,12 +16,15 @@ package {{package}};
...
 import {{resourceNameGlobalPackageName}}.ResourceNameType;
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
 
 // AUTO-GENERATED DOCUMENTATION AND CLASS
 @javax.annotation.Generated("by GAPIC protoc plugin")
 public class {{className}} {{extensionKeyword}} {{parentInterface}} {
 
   private static final String FIXED_VALUE = "{{fixedValue}}";
+  private static final Map<String, String> valueMap = ImmutableMap.of("FIXED_VALUE", FIXED_VALUE);
```
Comment (garrettjonesgoogle) [inline plugin/templates/resource_name_fixed.mustache]: Why not have it keyed with an empty string, like the untyped case?

Comment (andreamlin) [reply plugin/templates/resource_name_fixed.mustache]: Now keying on empty string; adjusted method comments to reflect that.

**Comment by andreamlin** [comment]
PTAL

**Comment by andreamlin** [comment]
https://github.com/googleapis/api-client-staging/pull/620/files has a green CI build with 
- the regen'd api-client-staging with all api-common dependencies bumped to 1.5.0 
- all Untyped*.java classes implementing the ResourceName methods, e.g. [UntypedDataSourceName.java](https://github.com/andreamlin/api-client-staging/blob/26488dd342aadbe81d9c3361da1496f01f0cb667/generated/java/proto-google-cloud-bigquerydatatransfer-v1/src/main/java/com/google/cloud/bigquery/datatransfer/v1/UntypedDataSourceName.java)
- the only fixed-value ResourceName subclass ([pubsub DeletedTopic.java](https://github.com/andreamlin/api-client-staging/blob/26488dd342aadbe81d9c3361da1496f01f0cb667/generated/java/proto-google-cloud-pubsub-v1/src/main/java/com/google/pubsub/v1/DeletedTopic.java)) implementing the ResourceName methods

**Pull request merged**